File: arrays_and_hashing/longest_consecutive_sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        if not nums:
            return 0

        s, best = set(nums), 0

        for v in s:
            if v - 1 not in s:
                curr = v
                loc = 1

                while curr + 1 in s:
                    curr += 1
                    loc += 1

                best = max(best, loc)

        return best

File: arrays_and_hashing/longest_consecutive_sequence.js
function longestConsecutive(nums) {
  if (nums.length === 0) return 0;

  let set = new Set(nums);
  let globalLongest = 0;

  for (const v of set) {
    if (!set.has(v - 1)) {
      let x = v; 
      let len = 1;

      while (set.has(x + 1)) {
        x++;
        len++;
      }

      if (len > globalLongest) globalLongest = len;
    }
  }

  return globalLongest
};